README.md

```diff
@@ -8,88 +8,115 @@
-## Instalation
+## Installation
 ``` bash
-# install dependency
 npm install --save vue2-smooth-scroll
 ```
 
-``` javascript
-// import on your project (less then 1KB gziped)
-import vueSmoothScroll from 'vue2-smooth-scroll'
-Vue.use(vueSmoothScroll)
+``` js
+import VueSmoothScroll from 'vue2-smooth-scroll'
+Vue.use(VueSmoothScroll)
 ```
 
-## Directive usage
+
+## Usage
+### Directive
 ``` html
-<a href="#div-id" v-smooth-scroll>Anchor</a>
-<div id="div-id"></div>
+<a href="#sec-3" v-smooth-scroll>Section 3</a>
+
+<section id="sec-3"></section>
 ```
 
-## Programmatic usage
+### Programmatic
 ``` js
-const exampleElement = this.$refs.exampleElement || this.$el || document.getElementById(...)
+const myEl = this.$refs.myEl || this.$el || document.getElementById(...)
+
 this.$smoothScroll({
-  scrollTo: exampleElement,
-  hash: '#exampleHash' // only required if updateHistory is true
-  ... // optional overrides for global config
+  scrollTo: myEl,
+  hash: '#sampleHash'  // required if updateHistory is true
 })
 ```
 
-## Custom options
-### Defaults
-``` js
-  {
-    duration: 500, // Animation speed
-    offset: 0, // Offset from element, you can use positive or negative values
-    container: '', // the scroll container, default is window,use document.querySelector to query the Element,
-    updateHistory: true //Push hash to history or not
-  }
-```
-### Example:
-``` html
-<div id="container">
-  <a href="#div-id" v-smooth-scroll="{ duration: 1000, offset: -50, container: '#container', updateHistory: false }">Anchor</a>
-  <div id="div-id"></div>
-</div>
-```
-
-
-### Without Browserify or Webpack
+### Direct in `<script>`
 ``` html
 <body>
   <div id="app">
-    <a href="#bottom" v-smooth-scroll>click me will scroll to bottom!</a>
+    <a href="#bottom" v-smooth-scroll>Let's go to #bottom!</a>
     <div style="height: 2000px;"></div>
     <span id="bottom">bottom</span>
   </div>
+  
   <script src="https://unpkg.com/vue/dist/vue.js"></script>
   <script src="https://unpkg.com/vue2-smooth-scroll"></script>
   <script>
-  var app = new Vue({
-    el: '#app',
-    data: {
-      message: 'Hello Vue!'
-    }
-  })
-</script>
+    var app = new Vue({
+      el: '#app'
+    })
+  </script>
 </body>
 ```
 
+
+## Custom options
+### Defaults
+``` js
+{
+  duration: 500,       // animation duration in ms
+  offset: 0,           // offset in px from scroll element, can be positive or negative
+  container: '',       // selector string or Element for scroll container, default is window
+  updateHistory: true  // whether to push hash to history
+}
+```
+
+### Global
+``` js
+import VueSmoothScroll from 'vue2-smooth-scroll'
+
+Vue.use(VueSmoothScroll, {
+  duration: 400,
+  updateHistory: false,
+})
+```
+
+### Directive:
+``` html
+<div id="container">
+  <a href="#div-id" v-smooth-scroll="{ duration: 1000, offset: -50, container: '#container' }">Anchor</a>
+  
+  <div id="div-id"></div>
+</div>
+```
+
+### Programmatic
+``` js
+this.$smoothScroll({
```

Comment:
I think it would be more appropriate in the `Usage` section above. Even better would be to add a demo link that will demonstrate the different options, so people can check-out the plugin without even installing.

When refactoring this README, I was thinking to add a demo link in a separate PR. What do you think?

Reply:
Actually I use [this](https://codesandbox.io/s/smooth-scroll-demo-i5sfd?file=/src/components/smooth.js) to do simply test. Maybe could reuse this?

Reply:
For basic initial demo, yeah. But without the file that contains the implementation of this package (`smooth.js`). It can be iterated on later with interactive fields to test different options (duration, offset, so on).